src/compiler/checker/lines_8000_12000.rs

```diff
@@ -161,19 +187,147 @@ impl TypeChecker {
         )
     }
 
-    pub(super) fn get_declared_type_of_class_or_interface(
+    pub(super) fn append_type_parameters(
+        &self,
+        mut type_parameters: Option<Vec<Rc<Type>>>,
+        declarations: &[Rc<Node>],
+    ) -> Option<Vec<Rc<Type>>> {
+        for declaration in declarations {
+            type_parameters = Some(append_if_unique(
+                type_parameters,
+                self.get_declared_type_of_type_parameter(
+                    self.get_symbol_of_node(&**declaration).unwrap(),
+                ),
+            ));
+        }
+        type_parameters
+    }
+
+    pub(super) fn get_outer_type_parameters(
```

Comment:
I think that "outer" type parameters might be like "closed-over" generic parameters from enclosing declarations?